identity-service/src/relay/txRelay.js

```diff
@@ -139,23 +145,61 @@ const sendTransactionInternal = async (req, web3, txProps, reqBodySHA) => {
       `L2 - txRelay - selected wallet ${wallet.publicKey} for sender ${senderAddress}`
     )
 
-    if (sendToNethermind) {
-      const ok = await relayToNethermind(encodedABI)
-      txParams = ok.txParams
-      txReceipt = ok.receipt
-    } else {
-      // use POA receipt as main receipt
-      const ok = await createAndSendTransaction(
-        wallet,
-        contractAddress,
-        '0x00',
-        web3,
-        req.logger,
-        gasLimit,
-        encodedABI
+    // send to POA
+    // PROD doesn't have sendToNethermindOnly and should default to POA
+    // STAGE defaults to nethermind but can send to POA when it has both addresses
+    const relayPromises = []
+    if (
+      !sendToNethermindOnly ||
+      (sendToNethermindOnly && nethermindContractAddress)
+    ) {
+      relayPromises.push(
+        createAndSendTransaction(
+          wallet,
+          contractAddress,
+          '0x00',
+          web3,
+          req.logger,
+          gasLimit,
+          encodedABI
+        )
+      )
+    }
+
+    // send to nethermind
+    // PROD doesn't have sendToNethermindOnly and only sends to nethermind when it has both addresses
+    // STAGE defaults to nethermind
+    if (
+      sendToNethermindOnly ||
+      (!sendToNethermindOnly && nethermindContractAddress)
+    ) {
+      if (!nethermindContractAddress) {
+        nethermindContractAddress = contractAddress
+        nethermindEncodedABI = encodedABI
+      }
```

Comment:
this is for staging which is on "nethermind" but may still have old clients that don't populate `nethermindContractAddress` and `nethermindEncodedABI`, right?

Reply:
yea if it's staging sendToNethermindOnly would be true and the primary payload would be the default nethermind one. this is actually on staging right now and relay should work the same